src/dynamic_fmt.c

```diff
@@ -404,6 +408,46 @@ char *RemoveHEX(char *output, char *input)
 	return output;
 }
 
+// Add $HEX$ encoding for salts where needed.  For canonicalized pot storage, we encode anything
+// containing unprintables or field separators, but never anything else
+static char *AddHEX(char *ciphertext) {
+	static char out[1024];
```

Comment:
How do we know(?) 1024 is enough?

Reply:
I think that's pretty much guaranteed for the dynamics but I see now this actually wasn't in the original function but an assumption of mine. I'll replace it with a `mem_alloc_tiny`